aws/lambda/benchmark_regression_summary_report/lambda_function.py

```diff
@@ -0,0 +1,467 @@
+        # set default
+        if not min_delta:
+            min_delta = dt.timedelta(days=2)
+
+        if not latest_ts_str:
+            return False
+        latest_dt = isoparse(latest_ts_str)
+        cutoff = end_time - min_delta
+
+        if latest_dt >= cutoff:
+            return True
+        logger.info(
+            "[%s] expect latest data to be after %s, but got %s",
+            config_id,
+            cutoff,
+            latest_dt,
+        )
+        return False
+
+    def _fetch_from_benchmark_ts_api(
+        self,
+        config_id: str,
+        end_time: dt.datetime,
+        start_time: dt.datetime,
+        source: BenchmarkApiSource,
+    ):
+        str_end_time = end_time.strftime("%Y-%m-%dT%H:%M:%S")
+        str_start_time = start_time.strftime("%Y-%m-%dT%H:%M:%S")
+        query = source.render(
+            ctx={
+                "startTime": str_start_time,
+                "stopTime": str_end_time,
+            }
+        )
+        url = source.api_query_url
+
+        logger.info("[%s]trying to call %s", config_id, url)
+        t0 = time.perf_counter()
+        try:
+            resp: BenchmarkTimeSeriesApiResponse = (
+                BenchmarkTimeSeriesApiResponse.from_request(url, query)
+            )
+
+            elapsed_ms = (time.perf_counter() - t0) * 1000.0
+            logger.info(
+                "[%s] call OK in %.1f ms (query_len=%d)",
+                config_id,
+                elapsed_ms,
+                len(query),
+            )
+            return resp.data
+        except requests.exceptions.HTTPError as e:
+            elapsed_ms = (time.perf_counter() - t0) * 1000.0
+            # Try to extract a useful server message safely
+            try:
+                err_msg = (
+                    e.response.json().get("error") if e.response is not None else str(e)
+                )
+            except Exception:
+                err_msg = (
+                    e.response.text
+                    if (e.response is not None and hasattr(e.response, "text"))
+                    else str(e)
+                )
+            logger.error(
+                "[%s] call FAILED in %.1f ms: %s", config_id, elapsed_ms, err_msg
+            )
+            raise
+
+        except Exception as e:
+            elapsed_ms = (time.perf_counter() - t0) * 1000.0
+            logger.error("[%s] call CRASHED in %.1f ms", config_id, elapsed_ms)
+            raise RuntimeError(f"[{config_id}]Fetch failed: {e}")
+
+    def _should_generate_report(
+        self,
+        cc: clickhouse_connect.driver.client.Client,
+        end_time: dt.datetime,
+        config_id: str,
+        f: Frequency,
+    ) -> bool:
+        def _get_latest_record_ts(
+            cc: clickhouse_connect.driver.Client,
+            config_id: str,
+        ) -> Optional[dt.datetime]:
+            table = BENCHMARK_REGRESSION_REPORT_TABLE
+            res = cc.query(
+                f"""
+                SELECT max(last_record_ts)
```

Comment:
Do you think that we should update the database schema to change the timestamp to epoch?  As I can see several benefits:

1. Simple epoch integer when using function like max
2. Easy to compute the start and end time difference without the need to use things like TimeDelta
3. Timezone gives me headache

We can convert epoch timestamp back to regular timestamp when generating the report.